src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const arrayActions = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      copyState = Object.assign({}, copyState, action.extraData);
+      arrayActions.push(copyState);
```

Comment:
don't push object to arrayActions in each `if` you can do it once at the end of the loop